File: module1/backend/Pocketbank.API/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Pocketbank.API.Models
{
    [Table("transactions")]
    public class Transaction
    {
        [Column("id")]
        public Guid Id { get; set; }
        
        [Column("account_id")]
        public Guid? AccountId { get; set; }
        
        [Column("category_id")]
        public Guid? CategoryId { get; set; }
        
        [Required]
        [Column("amount")]
        public decimal Amount { get; set; }
        
        [Required]
        [Column("transaction_date")]
        public DateTime TransactionDate { get; set; }
        
        [Column("description")]
        public string? Description { get; set; }
        
        [Column("created_at")]
        public DateTime? CreatedAt { get; set; }
        
        [Column("receipt_url")]
        public string? ReceiptUrl { get; set; }
        
        [Column("transaction_type")]
        public string? TransactionType { get; set; }
        
        // Navigation properties
        public virtual Category? Category { get; set; }
        public virtual Account? Account { get; set; }
    }
}

File: module1/frontend/src/types/transaction.ts
export interface User {
    id: string;
    email: string;
    name?: string;
    surname?: string;
    profilePictureUrl: string;
    passwordHash?: string;
    createdAt?: string;
}

export interface Category {
    id: string;
    userId?: string;
    name: string;
    color?: string;
    icon?: string;
    createdAt?: string;
}

export interface Transaction {
    id: string;
    accountId?: string;
    categoryId?: string;
    amount: number;
    transactionDate: string;
    description?: string;
    createdAt?: string;
    receiptUrl?: string;
    transactionType?: string;
    category?: Category;
    account?: Account;
}

export interface Account {
    id: string;
    userId?: string;
    accountName: string;
    accountType: string;
    balance?: number;
    currency?: string;
    createdAt?: string;
}

export interface RecurringTransaction {
    id: string;
    userId: string;
    description: string;
    amount: number;
    categoryId: string;
    accountId: string;
    startDate: string;
    frequency: string;
    isIncome: boolean;
    isActive: boolean;
    lastProcessed?: string;
    createdAt: string;
    category?: Category;
    account?: Account;
}

export interface Budget {
    id: string;
    userId?: string;
    name: string;
    amount: number;
    period: string;
    startDate: string;
    createdAt?: string;
}

export interface BudgetCategory {
    id: string;
    budgetId?: string;
    categoryId?: string;
    limit?: number;
    createdAt: string;
}
